src/main.cpp

```diff
@@ -0,0 +1,1605 @@
+
+    result = VK_ERROR_UNKNOWN;
+    scissor = {};
+    viewport = {};
+    begin_info = {};
+    render_pass_info = {};
+    clear_color = {{{0.0f, 0.0f, 0.0f, 1.0f}}};
+
+    begin_info.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
+    begin_info.flags = 0;
+    begin_info.pInheritanceInfo = nullptr;
+
+    result = vkBeginCommandBuffer(command_buffer, &begin_info);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to begin recording a command buffer");
+    }
+
+    render_pass_info.sType = VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO;
+    render_pass_info.renderPass = render_pass;
+    render_pass_info.framebuffer = swapchain_framebuffers[image_index];
+    render_pass_info.renderArea.offset = {0, 0};
+    render_pass_info.renderArea.extent = swapchain_extent;
+    render_pass_info.clearValueCount = 1;
+    render_pass_info.pClearValues = &clear_color;
+
+    viewport = get_viewport(swapchain_extent);
+    scissor = get_scissor(swapchain_extent);
+
+    vkCmdBeginRenderPass(
+            command_buffer,
+            &render_pass_info,
+            VK_SUBPASS_CONTENTS_INLINE);
+
+    vkCmdBindPipeline(
+            command_buffer,
+            VK_PIPELINE_BIND_POINT_GRAPHICS,
+            graphics_pipeline);
+
+    vkCmdSetViewport(command_buffer, 0, 1, &viewport);
+
+    vkCmdSetScissor(command_buffer, 0, 1, &scissor);
+
+    vkCmdDraw(command_buffer, 3, 1, 0, 0);
+
+    vkCmdEndRenderPass(command_buffer);
+
+    result = vkEndCommandBuffer(command_buffer);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to end command buffer");
+    }
+}
+
+static void
+draw_frame(
+        VkDevice& logical_device,
+        VkFence& fence_in_flight,
+        VkSwapchainKHR& swapchain,
+        VkSemaphore& semaphore_image_available,
+        VkSemaphore& semaphore_render_finished,
+        VkCommandBuffer& command_buffer,
+        VkRenderPass& render_pass,
+        std::vector<VkFramebuffer>& swapchain_framebuffers,
+        VkExtent2D& swapchain_extent,
+        VkPipeline& graphics_pipeline,
+        VkQueue& drawing_queue,
+        VkQueue& presentation_queue)
+{
+    VkResult result;
+    VkSubmitInfo submit_info;
+    VkPresentInfoKHR present_info;
+    VkSemaphore wait_semaphores[1];
+    VkSemaphore signal_semaphores[1];
+    VkPipelineStageFlags wait_stages[1];
+    VkSwapchainKHR swapchains[1];
+
+    uint32_t image_index;
+    uint32_t flags;
+
+    result = VK_ERROR_UNKNOWN;
+    submit_info = {};
+    present_info = {};
+    swapchains[0] = swapchain;
+    wait_semaphores[0] = semaphore_image_available;
+    signal_semaphores[0] = semaphore_render_finished;
+    wait_stages[0] = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+
+    flags = 0;
+    image_index = -1;
+
+    vkWaitForFences(logical_device, 1, &fence_in_flight, VK_TRUE, UINT64_MAX);
+
+    vkResetFences(logical_device, 1, &fence_in_flight);
+
+    // TODO handle return value
+    (void) vkAcquireNextImageKHR(
```

Comment:
Please handle return value.

Reply:
Added handling in: https://github.com/pottumuusi/neotetris/pull/1/commits/3e52caf15983b616c90201df55341a3002dbace2